### PR TITLE
NH-14155 Port KMS support from internal PoC to match refactored and updated upstream go-tuf

### PR DESCRIPTION
**Pull request by lootek**
see the original work by @pawel-kedzior-sw 
* https://github.com/solarwinds/go-tuf/pull/9
* https://github.com/solarwinds/go-tuf/pull/10

see the related upstream work
* https://github.com/theupdateframework/go-tuf/pull/148 (a massive discussion thread)
* https://github.com/theupdateframework/go-tuf/pull/126 (superseded by the above)
* https://github.com/theupdateframework/go-tuf/pull/248 (a rejected PR adding `import-key` feature)

### CODE REVIEW

File: cmd/tuf/main.go [hunk 1]
```diff
@@ -115,7 +122,16 @@ func runCommand(name string, args []string, dir string, insecure bool) error {
...
 		p = getPassphrase
 	}
-	repo, err := tuf.NewRepo(tuf.FileSystemStore(dir, p))
+
+	var s store.Store
+	switch manager {
+	case data.KeyStoreKMS.String():
+		s = store.KMSStore(dir, p)
+	default:
+		s = store.FileSystemStore(dir, p)
```
Comment (piotr-filipek-swi) [inline cmd/tuf/main.go:131]: Consider checking for `""` nad `"local"` as described in the usage - any other string should cause an error.

File: pkg/keys/kms.go [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+var (
+	initKMSServiceOnce sync.Once
+	kmsService         *kms.KMS
+)
+
+func KMSService() *kms.KMS {
+	initKMSServiceOnce.Do(func() {
+		region := os.Getenv("AWS_REGION")
+		if region == "" {
+			region = "us-east-1"
```
Comment (piotr-filipek-swi) [inline pkg/keys/kms.go:41]: Do we want to have a default ?

File: pkg/keys/kms.go [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+	keyUsageTypeSignVerify := kms.KeyUsageTypeSignVerify
+	createKeyInput.KeyUsage = &keyUsageTypeSignVerify
+
+	callerIdentityInput := &sts.GetCallerIdentityInput{}
+	callerIdentityOutput, err := stsService.GetCallerIdentity(callerIdentityInput)
+	if err != nil {
+		return nil, err
+	}
+
+	if policyJSON == "" {
```
Comment (piotr-filipek-swi) [inline pkg/keys/kms.go:113]: How about passing it as a file parameter in the command line?

File: pkg/keys/kms.go [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+				TagValue: aws.String(keyID),
+			},
+		},
+	}
+	_, err = kmsService.TagResource(tagInput)
+	if err != nil {
+		return nil, err
+	}
+
+	// err = addKeyEntry(&KMSEntry{
```
Comment (piotr-filipek-swi) [inline pkg/keys/kms.go:224]: Cleanup

File: pkg/keys/kms.go [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+	}
+
+	publicKey, err := x509.ParsePKIXPublicKey(output.PublicKey)
+	if err != nil {
+		panic(err)
+	}
+
+	publicKeyECDSA, isECDSA := publicKey.(*ecdsa.PublicKey)
+	if !isECDSA {
+		panic(fmt.Errorf("expected ecdsa public key but received %s", reflect.TypeOf(publicKeyECDSA)))
```
Comment (piotr-filipek-swi) [inline pkg/keys/kms.go:258]: Isn't RSA supported as well?

File: pkg/keys/kms.go (hunk 3 above)
Comment (lootek) [reply pkg/keys/kms.go:113]: nice idea, changed

File: pkg/keys/kms.go (hunk 5 above)
Comment (lootek) [reply pkg/keys/kms.go:258]: Good point... This is how it was coded in the PoC and I overlooked this issue.

**Comment by lootek** [comment]
too old stuff, needs to be addressed from scratch if/when needed

**Pull request closed without merging**